web/new_alt_text_manager.js

```diff
@@ -0,0 +1,396 @@
+/* Copyright 2023 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class NewAltTextManager {
+  #boundCancel = this.#cancel.bind(this);
+
+  #createAutomaticallyButton;
+
+  #currentEditor = null;
+
+  #cancelButton;
+
+  #descriptionContainer;
+
+  #dialog;
+
+  #disclaimer;
+
+  #firstTime = false;
+
+  #guessedAltText;
+
+  #isEditing = null;
```

Comment:
I would expect a variable starting with `is` to be a boolean instead of `null`.

Reply:
Yep I agree, but it was to be able to have something like `if (this.#isEditing !== value)...` the first time we set it.